hugo/assets/scripts/ast/statemachine.ts

```diff
@@ -0,0 +1,145 @@
+import { type } from "os";
+import { AstNode, LangiumAST } from "./ast-tools";
+
+
+
+export class StateMachineTools {
+    currentState: StateMachineState;
+    ast: StateMachineAstNode;   
+    constructor(ast: StateMachineAstNode) {
+        this.ast = ast;
+        this.currentState = this.getInitialState(ast);
+    }
+
+    /**
```

Comment:
Comments are actually very nice, but not when they reflect the name of the function.
You should talk about why is this piece of function like this and not what is obvious

Reply:
Since this is statemachine specific, you could add that it's the starting state for the statemachine, based on the AST.